test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -60,6 +62,20 @@ internal static void RunTest (MethodDeclarationSyntax m, List<AttributeSyntax> a
 								.Select (d => d.GetMessage ());
 			foreach (var attr in attrs) {
 				switch (attr.Name.ToString ()) {
+				case "ExpectedWarning":
+					Assert.True (
+						filtered.Any (mc => {
+							foreach (var attributeArgument in attr.ArgumentList!.Arguments) {
+								if (GetStringFromExpr (attributeArgument.Expression).StartsWith ("IL"))
+									continue;
+								if (!mc.Contains (GetStringFromExpr (attributeArgument.Expression)))
+									return false;
```

Comment:
I know this is tests, but this kind of thing is actually pretty slow. `GetStringFromExpr(...)` can be moved outside of all the loops, that should help a lot.